### PR TITLE
WIP - Refactor codebase to comply with C11

### PR DESCRIPTION
**Pull request by nabaco**
Replaced all auto_ptr with unique_ptr to comply with C11.
Relevant to #45

### CODE REVIEW

File: engine/src/gfx/radar/radar.cpp [hunk 1]
```diff
@@ -12,21 +12,21 @@
 namespace Radar
 {
 
-std::auto_ptr<Display> Factory(Type::Value type)
+std::unique_ptr<Display> Factory(Type::Value type)
```
Comment (BenjamenMeyer) [inline engine/src/gfx/radar/radar.cpp:15]: These might need to be `std::shared_ptr`; not sure. Does it run well with `std::unique_ptr`?

**Comment by LifWirser** [comment]
I just tried to clone and compile 0.5.3 only to get "bunches of "auto_ptr " errors when I have had none WTF happened. 0.5.3 had NONE before. did somebody know what they did, OR were they trying to impress somebody?

**Comment by nabaco** [comment]
> 
> 
> I just tried to clone and compile 0.5.3 only to get "bunches of "auto_ptr " errors when I have had none WTF happened. 0.5.3 had NONE before. did somebody know what they did, OR were they trying to impress somebody?

I think @royfalk can comment on this. To get rid of the warnings requires a single change in engine/CMakeList.txt, to remove the C11 standard from there.

**Comment by royfalk** [comment]
It sounds right. I just replied to a more recent email with a more full explanation.

**Comment by nabaco** [comment]
> 
> 
> So yeah. I think this PR was well-intentioned, but a little too simplistic. It's going to take a LOT more work to figure out which c++11/boost ptr types to use in each case. I started down this road myself, but would really like more input.
> 
> See, for example, #8 for further discussion.

Point taken. I thought there might be a quick way to solve the issue and remove all the warnings, but I guess it'll take some more time and effort. I'll close the PR.

**Comment by stephengtuggy** [comment]
Cool. Thanks, @nabaco .

**Pull request closed without merging**